dsc/src/resolve.rs

```diff
@@ -0,0 +1,152 @@
+    };
+
+    // try to deserialize the Include content as YAML first
+    let configuration: Configuration = match serde_yaml::from_str(&include_content) {
+        Ok(configuration) => configuration,
+        Err(_err) => {
+            // if that fails, try to deserialize it as JSON
+            match serde_json::from_str(&include_content) {
+                Ok(configuration) => configuration,
+                Err(err) => {
+                    return Err(format!("Error: Failed to read the configuration file '{include_path:?}' as YAML or JSON: {err}"));
+                }
+            }
+        }
+    };
+
+    // serialize the Configuration as JSON
+    let config_json = match serde_json::to_string(&configuration) {
+        Ok(json) => json,
+        Err(err) => {
+            return Err(format!("Error: JSON Error: {err}"));
+        }
+    };
+
+    let parameters = if let Some(parameters_file) = include.parameters_file {
+        // combine the path with DSC_CONFIG_ROOT
+        let parameters_file = normalize_path(Path::new(&parameters_file))?;
+        info!("Resolving parameters from file '{parameters_file:?}'");
+        match std::fs::read_to_string(&parameters_file) {
+            Ok(parameters) => {
+                let parameters_json = match parse_input_to_json(&parameters) {
+                    Ok(json) => json,
+                    Err(err) => {
+                        return Err(format!("Failed to parse parameters file '{parameters_file:?}' to JSON: {err}"));
+                    }
+                };
+                Some(parameters_json)
+            },
+            Err(err) => {
+                return Err(format!("Failed to resolve parameters file '{parameters_file:?}': {err}"));
+            }
+        }
+    } else {
+        debug!("No parameters file found");
+        None
+    };
+
+    Ok((parameters, config_json))
+}
+
+fn normalize_path(path: &Path) -> Result<PathBuf, String> {
+    if path.is_absolute() {
+        Ok(path.to_path_buf())
+    } else {
+        // check that no components of the path are '..'
+        if path.components().any(|c| c == std::path::Component::ParentDir) {
+            return Err(format!("Error: Include path must not contain '..': {path:?}"));
+        }
+
+        // use DSC_CONFIG_ROOT env var as current directory
+        let current_directory = match std::env::var(DSC_CONFIG_ROOT) {
+            Ok(current_directory) => current_directory,
+            Err(_err) => {
+                // use current working directory
+                match std::env::current_dir() {
+                    Ok(current_directory) => current_directory.to_string_lossy().into_owned(),
+                    Err(err) => {
+                        return Err(format!("Error: Failed to get current directory: {err}"));
+                    }
+                }
+            }
+        };
+
+        // combine the current directory with the Include path
+        Ok(Path::new(&current_directory).join(path))
```

Comment:
What if the path in the include file is a fully-specified one?

Reply:
This is already handled on line 127 above